tests/jballerina-unit-test/src/test/resources/test-src/types/regexp/regexp_type_test.bal

```diff
@@ -105,6 +106,22 @@ function testRegExpWithUserDefinedType() {
     assertEquality(re `AB*|[^c-d]{1,5}`, x7);
 }
 
+type T1 string:RegExp & readonly;
+type T2 regexp:RegExp & readonly;
+
+type Foo record {|
+    int e;
+    readonly regexp:RegExp f;
+|};
+
+Foo & readonly rf = {e: 1, f: re `test`};
+
+function testRegExpReadonlyLocalVars() {
+    string:RegExp & readonly x1 = re `test`;
+    T1 & readonly x2 = re `test`;
+    (T2 & readonly) & string:RegExp x3 = re `test`;
```

Comment:
Can use _ instead of the variable names.

Reply:
Updated in commit [#6c0d98](https://github.com/ballerina-platform/ballerina-lang/pull/41843/commits/6c0d989173879eca2bb63f3fa733c0c7ed2ee69d).